pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -186,13 +201,6 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 
 	i, err := ec2svc.CreateOrGetMachine(scope, bootstrapToken)
 	if err != nil {
-		if awserrors.IsFailedDependency(errors.Cause(err)) {
-			a.log.Error(err, "network not ready to launch instances yet")
-			return &controllerError.RequeueAfterError{
-				RequeueAfter: time.Minute,
-			}
-		}
-
```

Comment:
I think we want to check `isNodeJoin` here and clean up the config map if it's false.

Reply:
The cluster actuator has code to do this once the control plane is ready. Is that sufficient, or is there another case you're thinking of?

Reply:
No, I think cleaning up of the configMap in the Cluster actuator is correct, otherwise we have a race condition between the configMap being cleaned up and the controlplane ready annotation being set.

Reply:
It's in reference to the "invalid API request" mode above. I think that would take care of the "I asked for an IAM role that didn't exist" or "that security group is in a different account" cases. We make those kinds of mistakes fairly frequently in the inner iteration loop, and I'm not confident in future-me's ability to discover deleting the config map as the way to get myself un-stuck.

It's a little verbose, but maybe what I'm really asking for is instructions in the log message on line 181. Something to the effect of "if your control plane machine is stuck, run `kubectl delete -n NAMESPACE configmap/NAME-OF-CONFIGMAP` to try again"?

Reply:
Speaking of race conditions, I lost the read-modify-write cycle there 😄 

I was thinking of `GetOrCreateMachine` as create-only, but now that I've written that out I see my error. I guess we don't have the information at this point to determine whether it was a create error (could never possibly become ready) or a get error (it may yet be ready, who knows?), so it would indeed be racy.